### ISSUE TITLE
Global Blacklist for Series Based on Tags 

### ISSUE THREAD

**Issue by KenzKD** [mihonapp/mihon#1086]
### Describe your suggested feature

Have a Menu like a simplified version of the Categories Menu to Enter or Remove Keywords for the Blacklist ( Eg. Isekai, Smut, Action, Romance, Yaoi, etc).

Whenever a user opens any extension, any series with tags that match any Keywords in the Blacklist automatically get skipped over and are not shown to the User in the Popular, Latest or Search Screens. This way the User gets to view more relevant series and does not have to spend time scrolling through series which does not suite the Users taste.

If a user wants to view series in the blacklist they can either remove the Keyword or maybe there can be some toggle for it.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.5](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by mal-glitch** [comment]
Alternative Implementation Idea: Intercept at Selection Rather Than at Source Level
Instead of trying to filter results before they appear (which requires hooking into every source's pipeline), what if the blacklist worked at the point of selection?
The flow would be:
User taps a manga in the browse results
Before opening the manga page, the app reads the tags
If any tag matches the blacklist, instead of opening the page it shows a message like "Blacklisted tag" and returns to the previous screen
This approach requires no changes to any extensions, no per-source implementation, and has a single interception point in the codebase. The tradeoff is that blacklisted manga are still visible in browse results but can't be opened — which is good enough for most use cases.
This seems significantly easier to implement than full pre-filtering and could serve as a practical first version of the feature.

**Comment by KenzKD** [comment]
> Alternative Implementation Idea: Intercept at Selection Rather Than at Source Level Instead of trying to filter results before they appear (which requires hooking into every source's pipeline), what if the blacklist worked at the point of selection? The flow would be: User taps a manga in the browse results Before opening the manga page, the app reads the tags If any tag matches the blacklist, instead of opening the page it shows a message like "Blacklisted tag" and returns to the previous screen This approach requires no changes to any extensions, no per-source implementation, and has a single interception point in the codebase. The tradeoff is that blacklisted manga are still visible in browse results but can't be opened — which is good enough for most use cases. This seems significantly easier to implement than full pre-filtering and could serve as a practical first version of the feature.

This a slight improvement to having no blacklist but not that different from just reading the tags yourself. You can also more than likely guess the tags by just looking at the thumbnail. 

The whole point of having a black list is to reduce the clutter in browse results. I do agree that this would be more complicated but I think would be the overall better approach

**Comment by Adamantt777** [comment]
Yes, a blacklisting of tags is needed to hide unwanted thumbnails or manga covers, and only relevant results of the search appear. Many websites already have this by using a hyphen - before the unwanted tags in the search. Examples:  -yaoi  -scat
